go/x509util/x509util.go

```diff
@@ -0,0 +1,576 @@
+// Package x509util includes utility code for working with X.509
+// certificates from the x509 package.
+package x509util
+
+import (
+	"bytes"
+	"crypto/dsa"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rsa"
+	"encoding/hex"
+	"fmt"
+	"strconv"
+
+	"github.com/google/certificate-transparency/go/asn1"
+	"github.com/google/certificate-transparency/go/x509"
+	"github.com/google/certificate-transparency/go/x509/pkix"
+)
+
+//////////////////////////////////////
+// TODO(drysdale): export these constants/functions from x509 and remove this section.
+// Copy of unexported code from x509/x509.go
+var (
+	OidAttribute          = asn1.ObjectIdentifier{2, 5, 4}
+	OidCountry            = asn1.ObjectIdentifier{2, 5, 4, 6}
+	OidOrganization       = asn1.ObjectIdentifier{2, 5, 4, 10}
+	OidOrganizationalUnit = asn1.ObjectIdentifier{2, 5, 4, 11}
+	OidCommonName         = asn1.ObjectIdentifier{2, 5, 4, 3}
+	OidSerialNumber       = asn1.ObjectIdentifier{2, 5, 4, 5}
+	OidLocality           = asn1.ObjectIdentifier{2, 5, 4, 7}
+	OidProvince           = asn1.ObjectIdentifier{2, 5, 4, 8}
+	OidStreetAddress      = asn1.ObjectIdentifier{2, 5, 4, 9}
+	OidPostalCode         = asn1.ObjectIdentifier{2, 5, 4, 17}
+
+	OidPseudonym           = asn1.ObjectIdentifier{2, 5, 4, 65}
+	OidTitle               = asn1.ObjectIdentifier{2, 5, 4, 12}
+	OidDnQualifier         = asn1.ObjectIdentifier{2, 5, 4, 46}
+	OidName                = asn1.ObjectIdentifier{2, 5, 4, 41}
+	OidSurname             = asn1.ObjectIdentifier{2, 5, 4, 4}
+	OidGivenName           = asn1.ObjectIdentifier{2, 5, 4, 42}
+	OidInitials            = asn1.ObjectIdentifier{2, 5, 4, 43}
+	OidGenerationQualifier = asn1.ObjectIdentifier{2, 5, 4, 44}
+)
+
+var (
+	OidNamedCurveP224 = asn1.ObjectIdentifier{1, 3, 132, 0, 33}
+	OidNamedCurveP256 = asn1.ObjectIdentifier{1, 2, 840, 10045, 3, 1, 7}
+	OidNamedCurveP384 = asn1.ObjectIdentifier{1, 3, 132, 0, 34}
+	OidNamedCurveP521 = asn1.ObjectIdentifier{1, 3, 132, 0, 35}
+)
+
+// OidFromNamedCurve returns the OID used to specify the use of the given
+// elliptic curve.
+func OidFromNamedCurve(curve elliptic.Curve) (asn1.ObjectIdentifier, bool) {
```

Comment:
Yeah, it's kind of pointless -- it just indicates if the oid is non-nil.  This code is a direct copy of the [golang version](https://github.com/golang/go/blob/master/src/crypto/x509/x509.go#L511) but as it's unexported I guess there's no reason not to change the signature.